samples/bluetooth/peripheral/src/main.c

```diff
@@ -39,7 +39,7 @@ static struct bt_uuid_128 vnd_auth_uuid = BT_UUID_INIT_128(
 	0xf2, 0xde, 0xbc, 0x9a, 0x78, 0x56, 0x34, 0x12,
 	0x78, 0x56, 0x34, 0x12, 0x78, 0x56, 0x34, 0x12);
 
-static uint8_t vnd_value[] = { 'V', 'e', 'n', 'd', 'o', 'r' };
+static uint8_t vnd_value[] = "Vendor";
```

Comment:
If you want to add the NULL terminator to this value, then
1) Change the type to `char` instead of `uint8_t`
2) Modify the existing `sizeof` to `strlen` as well

Something needs to be changed at least; this mix of `strlen` og `sizeof` won't work properly

Reply:
It took me a while to understand what you meant so i'll clarify for everybody else.
There is a strlen in read_vnd.
There is a sizeof in write_vnd AND write_without_rsp_vnd.

The write functions sizeof in write_vnd and write_without_rsp_vnd does not append null terminator when written, so that should be fixed as well.

Another issue that i'm seeing is that there is currently 3 characteristics all pointing to the same value. That should be 3 independent values.

Reply:
I suggest we add a length variable that is used to check the current size, whereas `sizeof()` is used to check the max length of the string allowed.

Reply:
>There is currently 3 characteristics all pointing to the same value. That should be 3 independent values.

+1

>vnd_long_value also needs to be fixed.

Should be fixed in the same way IMHO